hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/dispatcher/Dispatcher.java

```diff
@@ -33,18 +34,24 @@ public class Dispatcher {
     private final Handlers handlers;
 
     private final StoreCache storeCache;
+    private final CryptoSignatureWaivers cryptoSignatureWaivers;
 
     /**
      * Constructor of {@code Dispatcher}
      *
      * @param handlers a {@link Handlers} record with all available handlers
      * @param storeCache a {@link StoreCache} that maintains stores for all active {@link
      *     HederaState}s
+     * @param cryptoSignatureWaivers the {@link CryptoSignatureWaivers} to use in pre-handle
      * @throws NullPointerException if one of the parameters is {@code null}
      */
-    public Dispatcher(@NonNull final Handlers handlers, @NonNull final StoreCache storeCache) {
+    public Dispatcher(
+            @NonNull final Handlers handlers,
+            @NonNull final StoreCache storeCache,
+            @NonNull final CryptoSignatureWaivers cryptoSignatureWaivers) {
```

Comment:
Do you think it is better to inject `PreHandleContext` here instead of `CryptoSignatureWaivers`?
So the needed `CryptoSignatureWaivers` or `FileSignatureWaivers` will be constructed for needed preHandle methods?

Reply:
Agreed, done.